sql/mysqld.cc

```diff
@@ -2960,9 +2950,17 @@ pthread_handler_t signal_hand(void *arg __attribute__((unused)))
 {
   sigset_t set;
   int sig;
-  my_thread_init();				// Init new thread
+
+  // thread is created, otherwise execution wouldn't reach this point
   signal_thread_in_use= 1;
 
+  // initialize this newly created thread
+  if (my_thread_init()) {
```

Comment:
This can fail only in one of two ways:
 - calloc fails (extremely unlikely)
 - my_thread_global_init_done is true at start of my_thread_init
I originally wrote this on 11.4 and ported it back to 10.5 before posting.  Now running on latest 11.4, there are new macOS test failures.  I cannot be sure that there was a real world failure, and the slow startup and 'hanging' install are both solved without this change.  From the code perspective, calling `my_end` (which calls `my_thread_global_end` and sets `my_thread_global_init_done` to 0) will cause `my_thread_init` to return with an error, leaving the thread started but not initialized.  In the server, we call `my_end` only once but in tests we call it in multiple places.  During initial development, I had an `abort()` in the body of this conditional to see if we had a case where the init would fail, and I'm certain I saw it, but now cannot replicate.

Reply:
* my_thread_global_init_done is true at start of my_thread_init

is false, you mean. And I don't see how it can happen. The server calls `my_init()` at the beginning of `mysqld_main()`

May be it's set back to false because of some early shutdown? This sounds more plausible, but I still couldn't find how it could happen